pkg/cmd/pr/checks/aggregate.go

```diff
@@ -0,0 +1,110 @@
+package checks
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/cli/cli/v2/api"
+)
+
+type check struct {
+	Name        string    `json:"name"`
+	State       string    `json:"state"`
+	StartedAt   time.Time `json:"startedAt"`
+	CompletedAt time.Time `json:"completedAt"`
+	Link        string    `json:"link"`
+	Bucket      string    `json:"bucket"`
+}
+
+type checkCounts struct {
+	Failed   int
+	Passed   int
+	Pending  int
+	Skipping int
+}
+
+func aggregateChecks(pr *api.PullRequest) ([]check, checkCounts, error) {
+	checks := []check{}
+	counts := checkCounts{}
+
+	if len(pr.StatusCheckRollup.Nodes) == 0 {
+		return checks, counts, fmt.Errorf("no commit found on the pull request")
+	}
+
+	rollup := pr.StatusCheckRollup.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
+	if len(rollup) == 0 {
+		return checks, counts, fmt.Errorf("no checks reported on the '%s' branch", pr.HeadRefName)
```

Comment:
@vilmibm I changed this from `pr.BaseRefName` to `pr.HeadRefName`, let me know if that makes sense. This is the only functionality outside of `--watch` mode that was changed.

Reply:
sure sounds more correct >_>